tests/integrationv2/common.py

```diff
@@ -232,6 +243,47 @@ class Curves(object):
     P384 = Curve("P-384")
 
 
+class Signature(object):
+    def __init__(self, name, min_protocol=Protocols.SSLv3, sig_type=None, sig_digest=None):
+        self.name = name
+        self.min_protocol = min_protocol
+        if 'RSA' in name:
+            self.algorithm = 'RSA'
+        if 'PSS' in name.upper():
+            self.algorithm = 'RSAPSS'
+        if 'EC' in name.upper() or 'ED' in name.upper():
+            self.algorithm = 'EC'
+
+        if '+' in name:
+            sig_type, sig_digest = name.split('+')
+
+        self.sig_type = sig_type
+        self.sig_digest = sig_digest
+
+    def __str__(self):
+        return self.name
+
+
+class Signatures(object):
+    RSA_SHA1 = Signature('RSA+SHA1')
```

Comment:
I'm not exactly following the logic here. You are sending this string 'RSA+SHA1' through a parsing function? Why do you need to do that? Where is the string originally from?

Reply:
OpenSSL uses the algorithm+hash notation to name a signature algorithm. I kept that notation for consistency.